src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -764,6 +780,8 @@ void KOSMAPOSLReconstruction<TargetT>::compute_kernelised_image(
       // Going to need the full emission regional normalised differences
       int dimf_row = this->num_voxels;
       int dimf_col = this->num_non_zero_feat-1;
+      
+      if(!this->iterative_kernel_image_fixed_sptr) 
```

Comment:
this might fail if you use the same object twice for reconstruction. It could still be non-zero from the previous run?

Reply:
mh I am confused about this. iterative_kernel_image_fixed_sptr points to a copy of the update only if we set "stop_kernel..." and we are in the exact iteration we chose. After that the pointer should not change as is meant to be "fixed". But I am probably missing something.

@ashgillman you mean to create a function for this condition? if (self->still_updating_iterative_kernel())

Reply:
> I am probably missing something.

My concern is what happens if you do (in Python or whatever).
```
recon = KOSMAPOSLReconstruction(...)
recon.set_input(...)
recon.reconstruct()
..
recon.set_input(...)
recon.reconstruct()
```
The 2nd recon would be affected with what you do before.

(It's quite different to think about interactive work vs command line)

Reply:
Mh I see. good point!